src/Illuminate/Validation/Rules/Password.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+namespace Illuminate\Validation\Rules;
+
+use Illuminate\Contracts\Validation\DataAwareRule;
+use Illuminate\Contracts\Validation\Rule;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Validator;
+
+class Password implements Rule, DataAwareRule
+{
+    /**
+     * If the password requires at least one uppercase and one lowercase letter.
+     *
+     * @var bool
+     */
+    protected $caseDiff = false;
+
+    /**
+     * The data under validation.
+     *
+     * @var array
+     */
+    protected $data;
+
+    /**
+     * If the password requires at least one letter.
+     *
+     * @var bool
+     */
+    protected $letters = false;
+
+    /**
+     * The minimum size of the password.
+     *
+     * @var int
+     */
+    protected $min = 8;
+
+    /**
+     * If the password should has not been compromised in data leaks.
+     *
+     * @var bool
+     */
+    protected $notCompromised = false;
+
+    /**
+     * If the password requires at least one number.
+     *
+     * @var bool
+     */
+    protected $numbers = false;
+
+    /**
+     * If the password requires at least one symbol.
+     *
+     * @var bool
+     */
+    protected $symbols = false;
+
+    /**
+     * The failure messages, if any.
+     *
+     * @var array
+     */
+    protected $messages = [];
+
+    /**
+     * Create a new rule instance.
+     *
+     * @param  int  $min
+     * @return void
+     */
+    public function __construct($min)
+    {
+        $this->min = max((int) $min, 1);
+    }
+
+    /**
+     * Set the data under validation.
+     *
+     * @param  array  $data
+     * @return $this
+     */
+    public function setData($data)
+    {
+        $this->data = $data;
+
+        return $this;
+    }
+
+    /**
+     * Sets the minimum size of the password.
+     *
+     * @param  int $size
+     * @return $this
+     */
+    public static function min($size)
+    {
+        return new static($size);
+    }
```

Comment:
Have you considered making the `min` a non-static function and adding a `__callStatic` that allows any of the modifiers (min, ensureNotCompromised, requireCaseDiff, requireLetters, requireNumbers, requireSymbols) to be the static call?

```php
// a password, requiring case diff with all the other settings defaulting (incl length 8)
Password::requireCaseDiff()
```

At the same time it allows for something like:
```php
$rule = new Password();

// a variable from the database, cache or config where the admin can enable certain password rules
if([some admin toggle enabled]) {
    $rule->min([some configurable variable])
}
```

Reply:
That doesn't work when methods are public. Try this:

```
class Foo
{
	function bar() {}
 
 	static function __callStatic($m, $args) {}
}

$foo = Foo::bar(); // Non-static method Foo::bar() cannot be called statically
```

Reply:
You're right. In that case I think I would replace the static function with a non-static and call them as `(new Password())->min(10)`.